mountpoint-s3-crt/src/http/request_response.rs

```diff
@@ -263,7 +274,7 @@ impl Iterator for HeadersIterator<'_> {
             let header = self
                 .headers
                 .get_index(self.offset)
-                .expect("HeadersIterator: failed to get next header");
+                .expect("headers at any offset smaller than original count should always exist given mut access");
```

Comment:
This is not true: look at how `Headers` implements `Clone`.

Reply:
Nothing to do with this change, but we should review this type.

Reply:
Good catch! I've opened issue https://github.com/awslabs/mountpoint-s3/issues/1209 to track.

As discussed offline, we'll go ahead with this change and track addressing the incorrect safety comments in a future change.